tracer/src/Datadog.Trace/Iast/StackWalker.cs

```diff
@@ -60,18 +46,65 @@ public static StackFrameInfo GetFrame()
         foreach (var frame in stackTrace.GetFrames())
         {
             var declaringType = frame?.GetMethod()?.DeclaringType;
+
+            foreach (var excludeType in ExcludeSpanGenerationTypes)
+            {
+                if (excludeType == declaringType?.FullName)
+                {
+                    return new StackFrameInfo(null, false);
+                }
+            }
+
             if (ExcludeSpanGenerationTypes.Contains(declaringType?.FullName))
             {
                 return new StackFrameInfo(null, false);
             }
 
             var assembly = declaringType?.Assembly.GetName().Name;
-            if (assembly != null && !AssemblyNamesToSkip.Contains(assembly))
+            if (assembly != null && !AssemblyExcluded(assembly))
             {
                 return new StackFrameInfo(frame, true);
             }
         }
 
         return new StackFrameInfo(null, true);
     }
+
+    public static bool AssemblyExcluded(string assembly)
+    {
+        if (ExcludedAssemblyCache.TryGetValue(assembly, out bool excluded))
+        {
+            return excluded;
+        }
+
+        excluded = IsExcluded(assembly);
+        ExcludedAssemblyCache[assembly] = excluded;
+
+        return excluded;
+    }
+
+    // For performance reasons, we are not supporting wildcards fully. We just need to use '*' at the end for now. We can use regular expressions
+    // if in the future we need a more sophisticated wildcard support
+    private static bool IsExcluded(string assembly)
+    {
+        foreach (var assemblyToSkip in AssemblyNamesToSkip)
+        {
+            if (assemblyToSkip.EndsWith("*"))
+            {
+                if (assembly.StartsWith(assemblyToSkip.Substring(0, assemblyToSkip.Length - 1), StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                if (assemblyToSkip.Equals(assembly, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
```

Comment:
I'm a bit late (as you've already merged it!) but given you're going to iterate this list every time you check an assembly, you're doing a lot of `Substring()` that I think you can avoid seeing as you aren't using "real" regex or wildcards, just prefix support.

1. Remove the `*` from the end of all the `AssemblyNamesToSkip`
2. Change the code here to the following:

```csharp
foreach (var assemblyToSkip in AssemblyNamesToSkip)
{
    if (assemblyToSkip.EndsWith('.'))
    {
        if (assembly.StartsWith(assemblyToSkip, StringComparison.OrdinalIgnoreCase))
        {
            return true;
        }
    }
    else
    {
        if (assembly.Equals(assemblyToSkip, StringComparison.OrdinalIgnoreCase))
        {
            return true;
        }
    }
}

return false;
```